config/request_types.go

```diff
@@ -0,0 +1,21 @@
+package config
+
+// RequestType : Request type enumeration
+type RequestType string
+
+// The request types (endpoints)
+const (
+	ReqTypeORTB2Web RequestType = "openrtb2-web"
+	ReqTypeORTB2App RequestType = "openrtb2-app"
+	ReqTypeAMP      RequestType = "amp"
+	ReqTypeVideo    RequestType = "video"
+)
+
+func RequestTypes() []RequestType {
```

Comment:
I agree with your point that `RequestType` in metrics shouldn't be used in core PBS code to make code decisions. However, I would recommend against moving `RequestType` to config as request types have nothing to do with configuration. `RequestType` in metrics were specifically defined for metric labels. The intention was not to use it in the core code but over time it has seeped into the core code. I'd recommend that we rather introduce something new in `exchange` that helps determine request type for decision in PBS core code or use signals that are already present in the request to infer the request type.